### DISCUSSION TITLE
How to add different vkd3d versions (vkd3d-proton)?

### DISCUSSION THREAD

**Discussion by stonebuckl** [bottlesdevs/Bottles#470] [Q&A] [answered]
Trying to run Diablo II Resurrected (D2R) from Battle.net through Bottles, I don't even get a window on Arch Linux. Judging from [these comments](https://github.com/HansKristian-Work/vkd3d-proton/issues/625) it should at least be possible to see a black window from the game (which seems much closer to a working game than an instant crash, as in my case). 

It seens like vkd3d-proton might be working better than the WineHQ version and I would like to try it out. However, when going to the Bottles preferences, I can only choose vkd3d-v2.4 and nothing else. Is there a way to also enable vkd3d-proton?

I installed the Battle.net client manually with the "Game" configuration of Bottles.

**Comment by mirkobrombin** [comment, marked as answer]
The vkd3d we provide is actually vkd3d-proton, just provided by lutris. Maybe you are facing runner problems. Which one are you using?

**Reply by stonebuckl** [reply]
I have tried several so far. Some of them show a window very briefly (basically flashing open and close), some don't open a window at all. The list I have tried so far:

- lutris-6.14-2
- vaniglia-6.13
- vaniglia-6.13-unstable
- Proton-6.13-GE-1
- Proton-6.14-GE-2

dxvk versions:
- dxvk-1.9
- dxvk-1.9.1

**Reply by stonebuckl** [reply]
Well, I'm not sure what the problem is but on another PC running Ubuntu (instead of Arch Linux) and an Nvidia GPU (instead of AMD) I can get the black screen with in-game sound using lutris-6.14-2 and dxvk-1.9.1. However, as soon as one would expect the game to show the main menu (after intro videos) the window crashes again.

EDIT: [This person](https://forums.lutris.net/t/has-anyone-had-luck-with-the-diablo-ii-resurrected-open-beta/13201) says that they can get the black window only with the lutris 6.0 runner. However, for me version 6.0 doesn't work either. For what it's worth, this is the console log from starting the game:

```
0154:err:winediag:SECUR32_initNTLMSP ntlm_auth was not found or is outdated. Make sure that ntlm_auth >= 3.0.25 is in your path. Usually, you can find it in the winbind package of your distribution.
0154:err:ntlm:ntlm_LsaApInitializePackage no NTLM support, expect problems
0154:err:module:map_image_into_view Could not map section .reloc, file probably truncated
0154:err:module:map_image_into_view Could not map section .reloc, file probably truncated
WARNING: radv is not a conformant vulkan implementation, testing use only.
0154:err:module:map_image_into_view Could not map section .reloc, file probably truncated
0154:err:module:map_image_into_view Could not map section .reloc, file probably truncated
WARNING: radv is not a conformant vulkan implementation, testing use only.
01d8:err:winediag:SECUR32_initNTLMSP ntlm_auth was not found or is outdated. Make sure that ntlm_auth >= 3.0.25 is in your path. Usually, you can find it in the winbind package of your distribution.
01d8:err:ntlm:ntlm_LsaApInitializePackage no NTLM support, expect problems
```

EDIT 2: Does Bottles include all these requirements [listed here](https://github.com/lutris/docs/blob/master/Battle.Net.md#additional-dependencies-required-specifically-for-battlenet)? I installed Bottles via Flatpak and I am really not sure what gets installed along with it.

**Reply by mirkobrombin** [reply]
I think it is a missing dependencies related problem. I'll come back to this discussion after today's release, in the meantime you can check out what is installed in the flatpak from its manifest: https://github.com/bottlesdevs/Bottles/blob/master/com.usebottles.bottles.json

**Reply by stonebuckl** [reply]
After using the [d3d12.dll provided by k0bin](https://github.com/HansKristian-Work/vkd3d-proton/pull/767#issuecomment-898939395) the game starts and runs with graphical glitches. However, it seems that this was actually the same issue even though I could not get the infamous black window. :)

EDIT: Graphical glitches are also fixed by running with RADV_DEBUG=nohiz set (conveniently done in Bottles)

**Reply by mirkobrombin** [reply]
So it is working using a specific version?

**Reply by stonebuckl** [reply]
Yes, it works with lutris-6.14-2, dxvk-1.9.1 and vkd3d-v2.4. Additionally, I manually replaced the d3d12.dll file and set it to `native` in the Bottles DLL Overrides configuration.
